### ISSUE TITLE
Kaniko integration test is flaky

### ISSUE THREAD

**Issue by dgageot** [GoogleContainerTools/skaffold#717]
```
    --- FAIL: TestRun/kaniko_example (1.23s)
4251
    	run_test.go:194: skaffold: Starting build...
4252
    		 Running [skaffold run]: stdout Starting build...
4253
    		, stderr: time="2018-06-20T16:28:27Z" level=fatal msg="build step: running kaniko build for gcr.io/k8s-skaffold/skaffold-example: creating kaniko pod: pods \"kaniko\" already exists"
4254
    		, err: exit status 1: exit status 1
4255
```

**Comment by dlorenc** [comment]
Was this part of CI? Or did you run it yourself?

**Comment by dgageot** [comment]
CI. I’ve seen it multiple times when I rebased my PRs

**Comment by dlorenc** [comment]
Ah, I think I know the issue. It's actually more a case that the kaniko integration itself is flaky. We have a hardcoded pod name right now, which means only one build can happen at a time per-namespace.

I mentioned this in https://github.com/GoogleContainerTools/skaffold/issues/671

We should use auto generated pod names here.

**Comment by dlorenc** [comment]
I think the fix here was actually #722 for later reference
